app/models/user.rb

```diff
@@ -48,4 +48,10 @@ def remove_all_roles(resource)
   def resource_roles(resource)
     roles.where(resource: resource)
   end
+
+  protected
```

Comment:
Is this `protected` because the method we're overriding is `protected` or for a different rationale? I don't have strong druthers about `protected`... though I personally avoid it if I can. That said, if we're using it to avoid changing the visibility of an overridden method, I'm fine with it. Approving either way.

Reply:
🤷 it's the visibility devise assigned to the method

Reply:
OK, that's what I assumed.